lessons/flow.md

```diff
@@ -112,21 +149,207 @@ No errors!
+  // ...
+}
+``` 
+
+`value` can be any sting and `undefined` but **not `null`**.
+
+### 3.2 [Mixed types](https://flow.org/en/docs/types/mixed/)
+
+In Flow we use Mixed types to indicates that a certain variable or argument can accept any kind of type. If you would like
+to interact with a variable which is `mixed` you need to ensure first that certain variable has a defined type.
+
+
+```flow js
+function stringify(value: mixed) {
+  if (typeof value === 'string') {
+    return "" + value; // Works!
+  } else {
+    return "";
+  }
+}
+```
+
+> Warning: Do not confused Mixed types with [`Any`](https://flow.org/en/docs/types/any/). **You should not use `Any`.** 
+
+### 3.3 [Function types](https://flow.org/en/docs/types/functions/)
+
+#### 3.3.1 [Return type of a function](https://flow.org/en/docs/types/functions/#toc-function-returns)
+
+We can add the return type of a function, by typing after the list of arguments.
+```flow js
+function foo(bar: string): number {
+  return 5;
+}
+```
+
+```flow js
+const method = (bar: string):number => {
+  return 5;
+}
+```
+
+#### 3.3.2 [Rest parameters](https://flow.org/en/docs/types/functions/#toc-rest-parameters)
+
+It is possible to type the rest parameters sintax as an `Array`.
+
+```flow js
+function method(...args: Array<number>) {
+  // ...
+}
+```
+
+####  3.3.3 [Arbitraty functions](https://flow.org/en/docs/types/functions/#toc-function-type)
+In the case you would like to express an arbitrary function, you should write `() => mixed`.
+
+> Warning: Do not use the `Function` type. Use `()=> mixed` instead. 
+
+
+#### 3.4 [Object Types](https://flow.org/en/docs/types/objects/)
+
+It is possible to define types for objects by defining the type of each of their fields:
+
+For example: 
+
+```flow js
+var obj: {
+  foo: number,
+  bar: boolean,
+  baz: string,
+} = {
+  foo: 1,
+  bar: true,
+  baz: 'three',
+};
+```
+
+In flow, it is considered safe to pass extra properties to a certain object of a certain type. Therefore,
+the following definition will work:
+
+```flow js
+var obj: {
+  foo: number,
+  bar: boolean,
+} = {
+  foo: 1,
+  bar: true,
+  baz: 'three',
+};
+```
+
+If we want to Exact match a object tyoe definition, we need to wrap the definition with `|`.
+
+```flow js
+var obj: {|
+  foo: number,
+  bar: boolean,
+|} = {
+  foo: 1,
+  bar: true,
+  baz: 'three',
+};
+```
+
+Flow will consider the definition above unsafe and it will fail in compile time.
+ 
+
+Read more about objects in the [flow docs](https://flow.org/en/docs/types/objects/).
+
+
+#### 3.5 [Union Types](https://flow.org/en/docs/types/unions/)
+
+Union types allow us to create a type which is one of a set of other types or constants. The definition 
+
+```flow js
+type Foo =
+  | Type1
+  | Type2
+  | ...
+  | TypeN
+```
+#### 3.6 [Array Types](https://flow.org/en/docs/types/arrays/)
+
+In order to create an array type it is possible to use `Array<Type>` type where `Type` is the type of elements in the array. 
```

Comment:
It might be worth hinting at generics here although you did say that you were going to miss them out. I just feel it might come up given the syntax here but maybe you can just wait for that in the lesson ...